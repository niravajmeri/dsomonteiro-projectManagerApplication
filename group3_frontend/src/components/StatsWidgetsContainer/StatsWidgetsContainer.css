.flex-container {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: flex-start;
    align-content: stretch;
    margin-top: 15px;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen,
        Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.number-widget {
    background-color: #ffffff;
    margin: 5px;
    flex-grow: 1;
    flex-basis: 30%;
    max-width: 50%;
}

.number-widget > .widget-title {
    font-size: 20px;
    margin: 10px;
    font-weight: bold;
}

.number-widget .widget-values .widget-unit {
    font-size: 15px;
    margin-left: 5px;
}

.number-widget .widget-values {
    text-align: center;
    line-height: 60px;
}
.number-widget .widget-values .widget-value {
    font-size: 60px;
}
