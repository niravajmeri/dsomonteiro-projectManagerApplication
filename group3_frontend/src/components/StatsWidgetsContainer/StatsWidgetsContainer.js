import React, { Component } from 'react';
import './StatsWidgetsContainer.css';
import StatsNumberWidget from '../StatsNumberWidget/StatsNumberWidget';

class StatsWidgetsContainer extends Component {
    renderWidgets = () => {
        return this.props.data.map((element, index) => {
            return (
                <StatsNumberWidget
                    key={index}
                    value={element.value}
                    title={element.title}
                    unit={element.unit}
                />
            );
        });
    };

    render() {
        return <div className="flex-container">{this.renderWidgets()}</div>;
    }
}

export default StatsWidgetsContainer;
