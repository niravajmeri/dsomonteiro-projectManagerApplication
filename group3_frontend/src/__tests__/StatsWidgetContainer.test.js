import React from 'react';
import { shallow } from 'enzyme';
import StatsWidgetsContainer from '../components/StatsWidgetsContainer/StatsWidgetsContainer';

describe('StatsWidgetsContainer', () => {
    it('should render correctly', () => {
        const dataArray = [
            { value: '1', title: 'task time', unit: 'hours' },
            { value: '2', title: 'project cost', unit: 'euros' }
        ];
        const component = shallow(<StatsWidgetsContainer data={dataArray} />);
        expect(component).toMatchSnapshot();
    });
});
