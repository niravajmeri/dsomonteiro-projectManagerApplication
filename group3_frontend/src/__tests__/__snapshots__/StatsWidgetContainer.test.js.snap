// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsWidgetsContainer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatsWidgetsContainer
    data={
        Array [
            Object {
              "title": "task time",
              "unit": "hours",
              "value": "1",
            },
            Object {
              "title": "project cost",
              "unit": "euros",
              "value": "2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StatsNumberWidget
          title="task time"
          unit="hours"
          value="1"
/>,
        <StatsNumberWidget
          title="project cost"
          unit="euros"
          value="2"
/>,
      ],
      "className": "flex-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "title": "task time",
          "unit": "hours",
          "value": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "title": "project cost",
          "unit": "euros",
          "value": "2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StatsNumberWidget
            title="task time"
            unit="hours"
            value="1"
/>,
          <StatsNumberWidget
            title="project cost"
            unit="euros"
            value="2"
/>,
        ],
        "className": "flex-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "title": "task time",
            "unit": "hours",
            "value": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "title": "project cost",
            "unit": "euros",
            "value": "2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
